### ISSUE TITLE
New way to find sponsored element

### ISSUE THREAD

**Issue by arye321** [tiratatp/facebook_adblock#132]
Hi, I wanted to make a pull request but the code is too complicated for me
so I'll suggest code improvement for finding sponsored elemnts

run this in facebook webpage console:

```javascript
document.querySelectorAll("use").forEach(element=>{
  const link = element.getAttribute("xlink:href").substring(1)
  if (document.querySelector(`text[id=${link}]`).innerHTML === "Sponsored"){
      element.closest('div[role]').style.borderStyle = 'solid'
      element.closest('div[role]').style.borderColor = 'red'
  }
})
```

this will mark all current ads in english interface . 

the new Sponsor label is inside an SVG with a `<use>` element, inside the `use` element theres a `shadow root (closed)` element.

![image](https://user-images.githubusercontent.com/9559551/191691192-96d9a358-c4a8-4c08-9c1d-28b65e7d3d79.png)

 you cant extract that so you have to follow the `xlink:href` of the `use`, which is always a `#gid(number)` , this leads to a `<text>` element

![image](https://user-images.githubusercontent.com/9559551/191692588-9e3eaeed-ff10-499c-9864-8c90bf604eff.png)

 the `innerHTML` of the `text` element is the text inside the original <use> element. if the innerHTML is 'Sponsored' you mark it.

hope it helps

**Comment by Sledmine** [comment]
This was not working for me, as it selects the entire div for where all the other posts live as well, I edited your selector to get the parent div the Sponsor post, it got dirty but it does the job, probably someone else can simplify it and integrate it with the extension:

```js
document.querySelectorAll("use").forEach(element=>{
  const link = element.getAttribute("xlink:href").substring(1)
  if (document.querySelector(`text[id=${link}]`).innerHTML === "Sponsored"){
      element.closest('div > div:not([class]) > div[class] > div:not([class]) > div[class]').style.borderStyle = 'solid'
      element.closest('div > div:not([class]) > div[class] > div:not([class]) > div[class]').style.borderColor = 'red'
  }
})
```
It manages to highlight just the Sponsored post (remember editing name of the text element depending on your language).
![image](https://user-images.githubusercontent.com/40512223/195990615-3d4a2ec9-0eb2-47d8-9137-750aef18bcfd.png)

**Comment by arye321** [comment]
> This was not working for me, as it selects the entire div for where all the other posts live as well, I edited your selector to get the parent div the Sponsor post, it got dirty but it does the job, probably someone else can simplify it and integrate it with the extension:

Right, they changed it a couple of days later, you can see my new method here:

https://github.com/arye321/fbab/blob/master/main.js

        useElements[len].closest(`div[class=""]`).hidden = true
